src/utils/util.ts

```diff
@@ -117,39 +117,50 @@ export const readPackageJson = async (config: d.Config, compilerCtx: d.CompilerC
   }
 };
 
-export const parsePackageJson = (
-  pkgJsonStr: string,
-  pkgJsonFilePath: string
-): { diagnostic: d.Diagnostic; data: d.PackageJsonData; filePath: string } => {
+/**
+ * A type that describes the result of parsing a `package.json` file's contents
+ */
+export type ParsePackageJsonResult = {
+  diagnostic: d.Diagnostic | null;
+  data: any | null;
+  filePath: string;
+};
+
+/**
+ * Parse a string read from a `package.json` file
+ * @param pkgJsonStr the string read from a `package.json` file
+ * @param pkgJsonFilePath the path to the already read `package.json` file
+ * @returns the results of parsing the provided contents of the `package.json` file
+ */
+export const parsePackageJson = (pkgJsonStr: string, pkgJsonFilePath: string): ParsePackageJsonResult | null => {
   if (isString(pkgJsonFilePath)) {
     return parseJson(pkgJsonStr, pkgJsonFilePath);
   }
   return null;
 };
 
-export const parseJson = (jsonStr: string, filePath?: string) => {
-  const rtn = {
-    diagnostic: null as d.Diagnostic,
-    data: null as any,
+/**
+ * Parse a string read from a `package.json` file
+ * @param jsonStr the string read from a `package.json` file
+ * @param filePath the path to the already read `package.json` file
+ * @returns the results of parsing the provided contents of the `package.json` file
+ */
+const parseJson = (jsonStr: string, filePath: string): ParsePackageJsonResult => {
+  const rtn: ParsePackageJsonResult = {
+    diagnostic: null,
+    data: null,
     filePath,
   };
 
-  if (isString(jsonStr)) {
```

Comment:
I removed this as the only caller of `parsePackageJson` validates that the argument it passes in when calling `parsePackageJson` is a string.  There's something to be said about being defensive here in our programming, but I tend to lean removing this in favor of the following considerations:
- the contents of the string aren't coming directly from a user (although its the contents of a file from a user provided path)
- leaning on the type system

the latter isn't probably sufficient on its own (at least not at this time) for user provided values, but I _think_ this is OK given the first point.  Feel free to argue the alternative though

Reply:
Yeah it's a little tricky here because `JSON.parse` called with a non-string will sometimes throw (e.g. w/ `JSON.parse({})` or `JSON.parse([])` and sometimes not (e.g. `JSON.parse(true)` or `JSON.parse(3)`). So a non-string value at runtime isn't a guarantee that the parsing will throw and that we'll return errors in the diagnostic and whatnot.

But I think that this is probably ok anyhow, I don't think that just checking it's a string gives us that much more safety anyway. We could still pass all kinds of things through that are valid JSON but which don't conform to what we'd expect a `package.json` to look like, e.g. `"3"`, `"true"`, `"[]"`, etc.